src/genetic_functions/select_parents.jl

```diff
@@ -6,23 +11,21 @@ function select_fitness_proportional_parents(population::Array{RuleNode}, fitnes
     return parent1, parent2
 end
 
-function select_two_random_parents(population::Array{RuleNode}, fitness_array::Array{<:Real})::Tuple{RuleNode,RuleNode}
-    sum_of_fitness = sum(fitness_array)
-    fitness_array_normalized = [fitness_value / sum_of_fitness for fitness_value in fitness_array]
-    parent1 = select_chromosome(population, fitness_array_normalized)
-    parent2 = select_chromosome(population, fitness_array_normalized)
-    return parent1, parent2
-end
 
+"""
+    select_chromosome(population::Array{RuleNode}, fitness_array::Array{<:Real})::RuleNode
+
+Selects a chromosome (individual) from the population based on a fitness array. The function uses a fitness-proportionate selection strategy, often referred to as "roulette wheel" selection. 
+"""
 function select_chromosome(population::Array{RuleNode}, fitness_array::Array{<:Real})::RuleNode
-    random_number = rand()        
+    random_number = rand()
     current_fitness_sum = 0
```

Comment:
Maybe we should mention here that `fitness_array` should already be normalized for the function to work.